### DISCUSSION TITLE
System Deployment Requirements

### DISCUSSION THREAD

**Discussion by jfuqua7** [serverpod/serverpod#189] [Q&A]
I'm looking to deploy an application using ServerPod on Linode. I would like to use Kubernetes. Do you have a recommended configuration setup and size of the Nodes for each? ServerPod, Postgres, Redis

Would also like your opinion on a high availability setup as well.

**Comment by vlidholt** [comment]
Unfortunately, I have no experience with Linode in particular. The latest version of Serverpod 0.9.8 comes with Terraform scripts for AWS that have a very reasonable setup. Perhaps you can use that as a starting point?

How powerful servers you need depends entirely on how many users you have and what your server is doing. For simpler tasks even the cheapest AWS tier works well.

**Reply by jfuqua7** [reply]
Cool. I will look at the AWS setup.

**Reply by jfuqua7** [reply]
I didn't see anything on Postgres and Redis in the AWS setup....did I miss it?

**Reply by vlidholt** [reply]
It sets up Postgres and Redis. This is a graph of what the deployment will look like (yes, I should probably clean that up):

![IMG_0024](https://user-images.githubusercontent.com/1539812/169662711-da6b043f-4b55-443f-9577-063af2139535.PNG)

If you are upgrading from a previous version of Serverpod it may be easiest to update Serverpod, then `serverpod create` a new project with the same name, but in a different folder. Copy over the `aws` directory and the `.github` directory from the new project. There is also a breaking change in 0.9.8, so you will need to update your config files (or copy them from the new project).

**Reply by jfuqua7** [reply]
Perfect...maybe I will just go ahead and run on AWS at first...I like Linode's fixed pricing.
I have not installed any versions yet. I am just starting up with the project.
